src/app/public/modules/flyout/flyout-adapter.service.ts

```diff
@@ -48,4 +56,23 @@ export class SkyFlyoutAdapterService {
       iframes[i].style.pointerEvents = enable ? '' : 'none';
     }
   }
+
+  public shouldClickCloseFlyout(host: ComponentRef<SkyFlyoutComponent>, target: Node): boolean {
+    if (host && host.location && !host.location.nativeElement.contains(target)) {
+
+      for (let otherHostTag of this.otherHostElementTags) {
+        let el = (<Element>target);
+        do {
+          if (el.tagName.toLowerCase() === otherHostTag) {
+            return false;
+          }
+          el = el.parentElement;
+        // tslint:disable-next-line:no-null-keyword
+        } while (el !== null && el.nodeType === 1);
```

Comment:
This does work, but I'm curious. Why are you using a do while loop over while loop? I wouldn't expect target to ever be null from a click event, but this just surprised me

Reply:
Truthfully it s because the sample code that I found on StackOverflow that I based this off of used a do-while. However, one thing that the do-while does give us is not checking the initial node for if it is an element (i.e. `nodeType === 1`). The click might have been on a "Text Node" and so we still will want to check that first one.

Reply:
Ah, alright cool :)
Thank you